manifest/protobuf.go

```diff
@@ -22,6 +24,53 @@ import (
 	"google.golang.org/protobuf/types/descriptorpb"
 )
 
+func getCacheDir() (string, error) {
+	cacheDir := filepath.Join("/tmp", "substreams-buf.cache")
+	return cacheDir, os.MkdirAll(cacheDir, 0755)
+}
+
+func generateCacheKey(module, version string, symbols []string) string {
+	h := sha256.New()
+	h.Write([]byte(module))
+	h.Write([]byte(version))
+	for _, symbol := range symbols {
+		h.Write([]byte(symbol))
+	}
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func loadFromCache(cacheKey string) (*descriptorpb.FileDescriptorSet, error) {
+	cacheDir, err := getCacheDir()
+	if err != nil {
+		return nil, err
+	}
+
+	cacheFile := filepath.Join(cacheDir, cacheKey+".pb")
+	data, err := os.ReadFile(cacheFile)
+	if err != nil {
+		return nil, err
+	}
+
+	fds := &descriptorpb.FileDescriptorSet{}
+	err = proto.Unmarshal(data, fds)
+	return fds, err
+}
+
+func saveToCache(cacheKey string, fds *descriptorpb.FileDescriptorSet) error {
+	cacheDir, err := getCacheDir()
+	if err != nil {
+		return err
+	}
+
+	data, err := proto.Marshal(fds)
+	if err != nil {
+		return err
+	}
+
+	cacheFile := filepath.Join(cacheDir, cacheKey+".pb")
```

Comment:
```suggestion
	cacheFile := filepath.Join(cacheDir, cacheKey+".file_descriptor.binpb")
```

The extra file_descriptor describes a bit the content, `binpb` is the Buf recognized extension for binary protobuf.